### DISCUSSION TITLE
Bluetooth tests latest version

### DISCUSSION THREAD

**Discussion by arenddeboer** [FabianHenneke/WearAuthn#38] [General]
In light of the request to test Bluetooth on the latest version, these are my findings.
OS: Ubuntu 24.04
Chrome: 131 64bit
Watch: Galaxy Watch 7 Ultra / Wear OS 5.0

Working great with Amazon AWS console, registering and logging tested. (https://aws.amazon.com/console/)

Trying to register on the below demo site doesn't work:
https://webauthn.io/
![issue](https://github.com/user-attachments/assets/ce7e49c0-37ee-4a26-b230-8eb520ead98e)
![watch-confirm](https://github.com/user-attachments/assets/d3d2ce9e-3ba3-4e5a-8be9-0c16f038067c)
![watch-ready](https://github.com/user-attachments/assets/6796f861-192c-49c4-b1f0-7f71f64f5239)

The dev console shows the following:

p Conditional UI
(index):693 AUTHENTICATION OPTIONS
(index):694 {
  "challenge": "9tXER9Rb-Qpx9zVbxmMU0mgayo46NLF--oj4RQmx_Nxww_44qeTZ-cooDRP3JLR1XhQO-vJ9POniubOuDVXOAw",
  "timeout": 60000,
  "rpId": "webauthn.io",
  "allowCredentials": [],
  "userVerification": "preferred"
}
(index):644 REGISTRATION OPTIONS
(index):645 {
  "rp": {
    "name": "webauthn.io",
    "id": "webauthn.io"
  },
  "user": {
    "id": "Km093UrRAkbEhA_w9zvgQTBegce9fChZZlGqHFFhVeo",
    "name": "watchbt",
    "displayName": "watchbt"
  },
  "challenge": "Txhwhg-ggypxVGKJOLFwLyuar2WoAs2IPolQijEaHpZpbGiidEqDcjyneuS_BeReGSksVRhSbLFFSGH2SsptEA",
  "pubKeyCredParams": [
    {
      "type": "public-key",
      "alg": -8
    },
    {
      "type": "public-key",
      "alg": -7
    },
    {
      "type": "public-key",
      "alg": -257
    }
  ],
  "timeout": 60000,
  "excludeCredentials": [],
  "authenticatorSelection": {
    "residentKey": "preferred",
    "requireResidentKey": false,
    "userVerification": "preferred"
  },
  "attestation": "none",
  "hints": [],
  "extensions": {
    "credProps": true
  }
}
(index):737 Conditional UI request was aborted

**Comment by arenddeboer** [comment]
The same error is shown when trying to register over Bluetooth on the following sites:
https://www.passkeys.io/
https://passkeys.guru/
https://www.token2.com/tools/fido2-demo
https://myaccount.google.com/

**Reply by arenddeboer** [reply]
On Windows 11 with either Chrome or Edge, when I select a security key to login with, it doesn't show a popup on the Watch while having the watch paired to Windows. Not sure exactly how this auth flow should work though. What I did:
- Pair watch with Windows 11 (it's showing connected on Watch and in Windows 11)
- Go to test website eg: https://www.token2.com/tools/fido2-demo
- Click register + various popups (see screenshots), but the Watch does not show a continue with this security key dialog.
- 
![Screenshot 2024-12-24 150611](https://github.com/user-attachments/assets/25b44ff9-0989-43a2-95be-d7c885d4ff86)
![Screenshot 2024-12-24 150628](https://github.com/user-attachments/assets/5e0fd916-229d-4335-9ad2-fd024734eabb)
![Screenshot 2024-12-24 150642](https://github.com/user-attachments/assets/9dc3841c-e3e1-48d3-bacc-eee3a2e63214)
![Screenshot 2024-12-24 150659](https://github.com/user-attachments/assets/6003fb53-619e-458b-8c19-2d1d3b894a90)
![Screenshot 2024-12-24 150715](https://github.com/user-attachments/assets/5919b4ba-1b8f-4fcd-b64c-31b8901af031)

**Reply by fmeum** [reply]
Thanks for the detailed test run, this is really helpful!

The Linux issue with only certain sites sounds like a logic issue that shouldn't be too difficult to fix. I hope that I can take a look soon.

Windows 10 supported WearAuthn pretty well, but I haven't personally tested it with Windows 11 yet. Since Windows Hello is closed source and changes pretty rapidly, I'm not sure whether we can realistically fix this issue.
